lib/rules/declaration-property-value-no-unknown/index.js

```diff
@@ -63,9 +65,47 @@ const rule = (primary, secondaryOptions) => {
 			return valuePattern && matchesStringOrRegExp(value, valuePattern);
 		};
 
-		const propertiesSyntax = (secondaryOptions && secondaryOptions.propertiesSyntax) || {};
+		const propertiesSyntax = {
+			// Take a shallow clone as this object will be appended to.
+			...((secondaryOptions && secondaryOptions.propertiesSyntax) || {}),
+		};
 		const typesSyntax = (secondaryOptions && secondaryOptions.typesSyntax) || {};
 
+		const typedCustomPropertyNames = new Map();
```

Comment:
[suggest] Giving type parameters makes the code a bit type-safer:

```suggestion
		/** @type {Map<string, string>} */
		const typedCustomPropertyNames = new Map();
```

Reply:
Good catch! 🙇